action/protocol/execution/evm/evm.go

```diff
@@ -292,7 +292,7 @@ func executeInEVM(evmParams *Params, stateDB *StateDBAdapter, hu config.HeightUp
 	if evmParams.contract == nil {
 		// create contract
 		var evmContractAddress common.Address
-		_, evmContractAddress, remainingGas, evmErr = evm.Create(executor, evmParams.data, remainingGas, evmParams.amount)
+		_, evmContractAddress, remainingGas, evmErr = evm.Create(executor, evmParams.data, evmParams.gas, evmParams.amount)
```

Comment:
it is not as simple as change `remainingGas` to `evmParams.gas`

intrinsicGas is first deducted as a base cost, so the evm is left to run with `remainingGas` amount of gas

Reply:
as you mentioned, "reasonably large contracts the deployment fails", for such case the solution is to increase the gas limit, the transaction will only be charged the gas amount that was actually consumed by EVM, so increasing gas limit does no harm